pkg/server/sandbox_status.go

```diff
@@ -38,32 +36,14 @@ func (c *criContainerdService) PodSandboxStatus(ctx context.Context, r *runtime.
 		return nil, fmt.Errorf("an error occurred when try to find sandbox: %v", err)
 	}
 
-	task, err := sandbox.Container.Task(ctx, nil)
-	if err != nil && !errdefs.IsNotFound(err) {
-		return nil, fmt.Errorf("failed to get sandbox container task: %v", err)
-	}
-
-	var pid uint32
-	var processStatus containerd.ProcessStatus
-	// If the sandbox container is running, treat it as READY.
-	if task != nil {
-		taskStatus, err := task.Status(ctx)
-		if err != nil {
-			return nil, fmt.Errorf("failed to get task status: %v", err)
-		}
-
-		pid = task.Pid()
-		processStatus = taskStatus.Status
-	}
-
 	ip := c.getIP(sandbox)
-	ctrInfo, err := sandbox.Container.Info(ctx)
-	if err != nil {
-		return nil, fmt.Errorf("failed to get sandbox container info: %v", err)
+	status := toCRISandboxStatus(sandbox.Metadata, sandbox.Status.Get(), ip)
+	if !r.GetVerbose() {
+		return &runtime.PodSandboxStatusResponse{Status: status}, nil
 	}
-	createdAt := ctrInfo.CreatedAt
-	status := toCRISandboxStatus(sandbox.Metadata, processStatus, createdAt, ip)
-	info, err := toCRISandboxInfo(ctx, sandbox, pid, processStatus, r.GetVerbose())
+
+	// Generate verbose information.
+	info, err := toCRISandboxInfo(ctx, sandbox)
```

Comment:
if r.GetVerbose() there is no need to get info here..

Reply:
If `r.GetVerbose()` it will return directly above.

Reply:
oh there it is :-)